### PR TITLE
✨🧪 Use injected AMP_EXP to toggle experiments

### PR DESCRIPTION
**Pull request by danielrozenberg**
Progress on #34013

### CODE REVIEW

**Comment by danielrozenberg** [comment]
@jridgewell so far this change will only work on the `AMP_EXP` object that gets injected into unversioned requests. What's the best way to fetch the JSON data from the `<script id=__AMP_EXP>` node?

File: src/experiments/index.js [hunk 1]
```diff
@@ -125,13 +125,18 @@ export function experimentToggles(win) {
...
-      if (typeof frequency === 'number' && frequency >= 0 && frequency <= 1) {
-        toggles[experimentId] = Math.random() < frequency;
-      }
+  // eslint-disable-next-line local/window-property-name
+  win.AMP_EXP ??= JSON.parse(win.__AMP_EXP?.textContent || '{}');
+
+  // Read default and injected configs of this build.
+  const buildExperimentConfigs = {
+    ...(win.AMP_CONFIG ?? {}),
+    ...(win.AMP_EXP ?? {}),
```
Comment (jridgewell) [inline src/experiments/index.js:134]: Due to the `??=` above, we've can assert that `AMP_EXP` is defined on `win`.

```suggestion
    ...win.AMP_EXP,
```

**Review by jridgewell — APPROVED** [review]

File: src/experiments/index.js (hunk 1 above)
Comment (danielrozenberg) [reply src/experiments/index.js]: Oops didn't notice it was a commit suggestion... pushed it in a separate commit

**Comment by danielrozenberg** [comment]
@jridgewell @rileyajones made a few type-checking changes without actually knowing much about how we do type checking, so I would very much appreciate an extra look

**Review by jridgewell — APPROVED** [review]

**Review by rileyajones — APPROVED** [review]
Typing looks good

**Comment by rileyajones** [comment]
I think this is leading to build failures
https://app.circleci.com/pipelines/github/ampproject/amphtml/11002/workflows/e68a853c-d2bf-4f79-ab19-ea6fc83ff4a3/jobs/173299

File: build-system/tasks/check-types.js [hunk 2]
```diff
@@ -108,7 +108,10 @@ const TYPE_CHECK_TARGETS = {
   'src-context': ['src/context/**/*.js', ...CORE_SRCS_GLOBS],
   'src-core': CORE_SRCS_GLOBS,
   'src-examiner': ['src/examiner/**/*.js'],
-  'src-experiments': ['src/experiments/**/*.js', ...CORE_SRCS_GLOBS],
+  'src-experiments': {
+    srcGlobs: ['src/experiments/**/*.js', ...CORE_SRCS_GLOBS],
+    externGlobs: ['build-system/externs/amp.extern.js'],
```
Comment (samouri) [inline build-system/tasks/check-types.js:113]: I think @rcebulko's intention is to not include build-system (BS) externs in any of these folder.

cc @rcebulko

Comment (danielrozenberg) [reply build-system/tasks/check-types.js:113]: I'll add Ryan as reviewer on the un-revert PR after #34796

Comment (rcebulko) [reply build-system/tasks/check-types.js:113]: That is correct 👍🏻

Comment (rcebulko) [reply build-system/tasks/check-types.js:113]: Related: https://github.com/ampproject/amphtml/pull/34797

**Pull request merged**